tool/test/org/antlr/v4/test/tool/BaseTest.java

```diff
@@ -112,6 +113,7 @@
 import static org.junit.Assert.assertNotNull;
 import static org.junit.Assert.assertTrue;
 
+@SuppressWarnings("resource")
```

Comment:
:thought_balloon: According to [this StackOverflow answer](http://stackoverflow.com/a/1206001/138304), this message only applies to resources of type `Closeable`. We don't have any resources of type `Closeable` because it was first introduced in Java 7; Eclipse should be picking up this configuration but appears to not be. I'll look into this a bit more and try to automate Eclipse's selection of `-source 1.6 -target 1.6`. I'm expecting this to work even for users working with JDK 7 or JDK 8, but I haven't figured everything out yet.

I think we should only include this use of `@SuppressWarnings` if we can't get things to work using an updated build configuration that accurately reflects the limitations of our current compatibility goals.